app/db/crud/document.py

```diff
@@ -237,3 +240,42 @@ def _get_documents_for_family_import_id(
     ]
 
     return documents
+
+
+class DocumentExtraCache:
+    """A simple cache for document -> family info mapping details."""
+
+    def __init__(self):
+        self._ttl = timedelta(minutes=60)
+        self._timestamp = datetime.utcnow() - self._ttl
+        self._doc_extra_info: Mapping[str, Mapping[str, str]] = {}
+
+    def get_document_extra_info(self, db: Session) -> Mapping[str, Mapping[str, str]]:
+        """
+        Get a map from document_id to useful properties for processing.
+
+        :param [Session] db: Database session to query
+        :return [Mapping[str, Mapping[str, str]]]: A mapping from document import_id to
+            document slug, family slug & family import id details.
+        """
+        if datetime.utcnow() - self._timestamp >= self._ttl:
+            self._doc_extra_info = self._update_document_extra(db)
+            self._timestamp = datetime.utcnow()
+        return self._doc_extra_info
+
+    def _update_document_extra(self, db: Session) -> Mapping[str, Mapping[str, str]]:
```

Comment:
Would `_query_doc_extra_info` be a better name?

Reply:
done in next push